scripts/redux/actions.js

```diff
@@ -420,6 +425,105 @@ const subscribeActions = {
   }
 };
 
+const messaging = firebase.messaging();
+messaging.onMessage(({ notification }) => {
+  toastActions.showToast({
+    message: `${notification.title} ${notification.body}`,
+    action: {
+      title: '{$ notifications.toast.title $}',
+      callback: () => {
+        routingActions.setLocation(notification.click_action);
+      }
+    }
+  })
+});
+messaging.onTokenRefresh(() => {
+  notificationsActions.getToken(true);
+});
+const notificationsActions = {
+  requestPermission: () => {
+    return messaging.requestPermission()
+      .then(() => {
+        notificationsActions.getToken(true);
+      })
+      .catch(error => {
+        console.log('Unable to get permission to notify.', error);
+        store.dispatch({
+          type: UPDATE_NOTIFICATIONS_STATUS,
+          status: NOTIFICATIONS_STATUS.DENIED
+        });
+      });
+  },
+
+  getToken: subscribe => {
+    return messaging.getToken()
+      .then(currentToken => {
+        if (currentToken) {
+          const state = store.getState();
+          const subscribersRef = firebase.database().ref(`/notifications/subscribers/${currentToken}`);
+          const subscribersPromise = subscribersRef.once('value');
+
+          const userUid = state.user && state.user.uid ? state.user.uid : null;
+          let userSubscriptionsPromise = Promise.resolve(null);
+          let userSubscriptionsRef;
+          if (userUid) {
+            userSubscriptionsRef = firebase.database().ref(`/notifications/users/${userUid}/${currentToken}`);
+            userSubscriptionsPromise = userSubscriptionsRef.once('value');
+          }
+
+          Promise.all([subscribersPromise, userSubscriptionsPromise])
+            .then(results => {
+              const isSubscribed = results[0].val();
+              const isUserSubscribed = results[1] ? results[1].val() : false;
```

Comment:
what is the difference between isSubscribed and isUserSubscribed?

Reply:
isSubscribed - for anonymous users
isUserSubscribed - authenticated

Reply:
Then I suggest to rename them like
isAnonymousUserSubscribed
isAuthenticatedUserSubscribed

Or any other your variant, just to make code more obvious